### PR TITLE
feat: close companion on escape key

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Set the companion in collapsed form when `esc` key is pressed.
- Should not close when any of our modal is open. 

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

DD-{number} #done

### CODE REVIEW

File: packages/extension/src/companion/CompanionMenu.tsx [hunk 1]
```diff
@@ -149,6 +150,20 @@ export default function CompanionMenu({
 
   const tooltipContainerProps = { className: 'shadow-2 whitespace-nowrap' };
 
+  const onsEscape = () => {
```
Comment (rebelchris) [inline packages/extension/src/companion/CompanionMenu.tsx:153]:
```suggestion
  const onEscape = () => {
```

File: packages/extension/src/companion/CompanionMenu.tsx [hunk 2]
```diff
@@ -149,6 +150,20 @@ export default function CompanionMenu({
...
+    if (!companionState) {
+      return;
+    }
+
+    trackEvent({ event_name: 'close companion' });
+    toggleCompanionExpanded({ companionExpandedValue: false });
+    setCompanionState(false);
+  };
+
+  useKeyboardNavigation(window, [['Escape', onsEscape]], {
```
Comment (rebelchris) [inline packages/extension/src/companion/CompanionMenu.tsx:163]:
```suggestion
  useKeyboardNavigation(window, [['Escape', onEscape]], {
```

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice one!

**Review by idoshamun — APPROVED** [review]

**Comment by sshanzel** [comment]
@rebelchris fixed the typos. Thank you for the suggested commits! Let me know if you have further concerns 🚢

**Comment by rebelchris** [comment]
@sshanzel All good, ship it 😂⛴

**Review by rebelchris — APPROVED** [review]

**Pull request merged**